File: src/cviz/timeline/commands/StopCommand.java
package cviz.timeline.commands;

import cviz.ITimeline;
import cviz.timeline.Trigger;

import java.util.List;
import java.util.stream.Collectors;

public class StopCommand extends AmcpCommand {
    public StopCommand(int layerId, String command) {
        super(layerId, command);
    }

    @Override
    public boolean execute(ITimeline timeline) {
        List<Trigger> oldTriggers = timeline.getActiveTriggers().stream()
                .filter(t -> t.isLoop() && t.getLayerId() == getLayerId())
                .collect(Collectors.toList());
        timeline.getActiveTriggers().removeAll(oldTriggers);

        return super.execute(timeline);
    }

    @Override
    public String getCommandName() {
        return "StopCommand";
    }
}

File: CViz/Timeline/Command/StopCommand.cs
using CViz.Timeline.Triggers;

namespace CViz.Timeline.Command
{
    class StopCommand : AmcpCommand
    {
        public StopCommand(int layerId, string command) : base(layerId, command)
        {
        }

        public override bool Execute(ITimeline timeline)
        {
            timeline.RemoveAllTriggers(t => t is LoopTrigger && t.Layer == LayerId);

            return base.Execute(timeline);
        }

        protected override string CommandName => "StopCommand";
    }
}